mito-ds/monorepo
# mitosheet: add id, wrap text on code snippets

# Description

- When the `MITO_CONFIG_CODE_SNIPPETS_VERSION` and `MITO_CONFIG_CODE_SNIPPETS_URL` are set, uses them to  retrieve the code snippets and displays them in the taskpane
- Adds a Get Support option to each code snippet dropdown that can be configured through the `MITO_CONFIG_CODE_SNIPPETS_SUPPORT` environment variable
- In order to type check the requests code, I installed `types-requests`. 
- Fixes race condition in the code snippets taskpane where clicking on another code snippet while a code snippet was already open resulted in no dropdowns being open. 

# Testing

Set the environment variables and test them. Here's some starter code
```
import os
os.environ['MITO_CONFIG_VERSION'] = '2'
os.environ['MITO_CONFIG_SUPPORT_EMAIL'] = 'aaron@sagacollab.com'
os.environ['MITO_CONFIG_CODE_SNIPPETS_SUPPORT_EMAIL'] = 'naterush1997@gmail.com'
os.environ['MITO_CONFIG_CODE_SNIPPETS_VERSION'] = '1'
os.environ['MITO_CONFIG_CODE_SNIPPETS_URL'] = 'json_test'
```

- make sure that the get_code_snippets taskpane uses the environment variables to get the code snippets instead of the default code snippets.
- make sure that clicking the get support button generates an email instead of sending users to slack. 

# Documentation

Yes, we should add code snippets to the docs
